packages/service-library/src/servicelib/async_utils.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+from functools import wraps
+from collections import deque
+import attr
+from typing import List, Dict
+
+
+@attr.s
+class Context:
+    in_queue: asyncio.Queue = attr.ib()
+    out_queue: asyncio.Queue = attr.ib()
+    initialized: bool = attr.ib()
+
+
+def run_serialized_in_context(target_args: List[str] = None):
+    """All request to function with same calling context will be serialized.
+
+    Example:
+
+    Given the following decorated function
+
+        @run_serialized_in_context(target_args=["param3", "param1"])
```

Comment:
I find confusing the term "serialized" in this context. If I understand it right, you mean that the function calls will run sequentially (vs in parallel), i.e. one after the other.  The term serialization for me suggest a mechanism persist/store data. 

If you refer to the opposite of "parallel" I would suggest "sequencial" instead

Am I missing something? :-)

Reply:
I was using a false friend, I will rename it to "sequencial".

Reply:
ok but then use "sequential"